composer/datasets/streaming/download.py

```diff
@@ -94,7 +96,21 @@ def dispatch_download(remote, local, timeout: float):
         local (str): Local path (local filesystem).
         timeout (float): How long to wait for file to download before raising an exception.
     """
-    if os.path.exists(local):
+
+    def dispatch_decompress(local, local_decompressed, compression_scheme):
```

Comment:
Can this function be a private helper, rather than being defined within a function? Also needs type annotations

Also, does this need to be a function, or can it be hardcoded on line 127? I think it is only used in one place?